### ISSUE TITLE
Failure to Initialize (from source)

### ISSUE THREAD

**Issue by alexfornuto** [lando/lando#395]
Bug Report
==========

**Tell us about your setup**

Antergos (Arch) Linux, 4.13.4-1-ARCH 
Lando v3.0.0-beta.19 (I've tried from scratch on both the `master` and `real-mccoy` branches, same result)
node v6.11.3
npm 5.4.2

**Tell us about your `.lando.yml`**

I'm unable to create the `.lando.yml` file, hence the issue :smile:.

**Tell us about the command you were running**

I'm attempting to initialize Lando on an existing site from Pantheon. While in the site's local repo folder I'm running:

```
lando.dev init --recipe pantheon
```

`lando.dev` is a symlink to `/home/[USER]/repos/lando/bin/lando.js`, per your docs.

**Tell us about the error you got**

```
➜  # lando.dev init --recipe pantheon --verbose
? Choose a Pantheon account [ACCOUNT]
? Which site? [SITENAME]
warn: Attempting to start engine...
[sudo] password for [USER]: 
error:  Error: Error: Error: Error: 1
    at /home/[USER]/repos/lando/lib/daemon.js:135:11
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at Object.exports.up (/home/[USER]/repos/lando/lib/daemon.js:134:9)
    at /home/[USER]/repos/lando/lib/engine.js:161:19
From previous event:
    at Object.exports.up (/home/[USER]/repos/lando/lib/engine.js:160:4)
    at /home/[USER]/repos/lando/lib/engine.js:98:22
From previous event:
    at verifyDaemonUp (/home/[USER]/repos/lando/lib/engine.js:96:4)
From previous event:
    at verifyDaemonIsReady (/home/[USER]/repos/lando/lib/engine.js:112:4)
    at Object.exports.exists (/home/[USER]/repos/lando/lib/engine.js:343:10)
    at Object.kill (/home/[USER]/repos/lando/plugins/lando-core/lib/init.js:177:25)
    at /home/[USER]/repos/lando/plugins/lando-core/lib/tasks/init.js:119:27
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at Object.run (/home/[USER]/repos/lando/plugins/lando-core/lib/tasks/init.js:118:8)
    at /home/[USER]/repos/lando/lib/tasks.js:205:21
From previous event:
    at /home/[USER]/repos/lando/lib/tasks.js:204:8
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
From previous event:
    at Object.command.handler (/home/[USER]/repos/lando/lib/tasks.js:189:6)
    at Object.self.runCommand (/home/[USER]/repos/lando/node_modules/yargs/lib/command.js:170:22)
    at parseArgs (/home/[USER]/repos/lando/node_modules/yargs/yargs.js:920:28)
    at Object.get [as argv] (/home/[USER]/repos/lando/node_modules/yargs/yargs.js:860:16)
    at /home/[USER]/repos/lando/lib/cli.js:267:7
From previous event:
    at Object.exports.init (/home/[USER]/repos/lando/lib/cli.js:237:4)
    at /home/[USER]/repos/lando/bin/lando.js:21:20
From previous event:
    at Object.<anonymous> (/home/[USER]/repos/lando/bin/lando.js:20:2)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:502:3

```

**Tell us generally about your bug**

I understand that I'm working in a non-standard environment, so if you close this issue as "out of scope" I won't rage. I'm submitting this issue in the hopes that it might help make Lando more accessible.

I've tried this with both the latest node, and with the LTS version specified in your docs, with the same result. After getting the error above, I rm'd the lando folder, cloned again, switched to `real-mccoy`, which I assumed was your code name for the stable branch, and ran `npm install` again.

**Tell us more**

Could better documentation have prevented this issue?
> Perhaps by documenting how to access the stable vs development branches?

Does this bug prevent you from using lando?
> Indeedy do.

**Comment by pirog** [comment]
@alexfornuto it looks you don't have docker up and running

**Comment by alexfornuto** [comment]
I'm running Docker version 17.09.0-ce, build afdb6d44a8, and I just manually started docker.socket and docker.service in systemd, with the same result.

**Comment by pirog** [comment]
@alexfornuto I also assume you installed `docker-compose` (https://docs.devwithlando.io/installation/installing.html#from-source) and moved it into the correct location?

So, `lando` expects two things on linux which might not be consistent across all distros. If these are different for you let me know and we can provide some logic to handle your distro

1. Start up command - `sudo service docker start` (https://github.com/lando/lando/blob/master/lib/daemon.js#L29)
2. Socket location - `/var/run/docker.sock`
(https://github.com/lando/lando/blob/master/lib/daemon.js#L206)

If you've got `docker` already running then the second one is the only critical one. I'd try editing the code with your socket location and seeing if that does the trick.

**Comment by alexfornuto** [comment]
OK, I've solved it thanks to yours and @dustinleblanc's help. 

1) My user was not a member of the `docker` group. A little `usermod -aG docker $USER` with a side of logout-login fixed that.

2) I could have sworn up and down that I installed docker-compose. But I'm a big idiot who didn't. Installed, and symlinked to `/usr/share/lando/bin/` FTW.

I'll submit a PR to the referenced guide, documenting some of the gotcha's I encountered.
